src/predicates.js

```diff
@@ -134,23 +132,23 @@ function messageIsIgnored(logger) {
         return true;
       }
 
-      body = item.body;
-      traceMessage = _.get(body, 'trace.exception.message');
-      bodyMessage = _.get(body, 'message.body');
+      messages = messagesFromItem(item);
 
-      exceptionMessage = traceMessage || bodyMessage;
-
-      if (!exceptionMessage){
+      if (messages.length === 0){
         return true;
       }
 
       len = ignoredMessages.length;
       for (i = 0; i < len; i++) {
         rIgnoredMessage = new RegExp(ignoredMessages[i], 'gi');
-        messageIsIgnored = rIgnoredMessage.test(exceptionMessage);
 
-        if (messageIsIgnored) {
-          break;
+        var messagesLength = messages.length;
+        for (j = 0; j < messagesLength; j++) {
```

Comment:
All local variables are declared in the function header. I would shift the `messageLength` declaration up.

However, the `messageLength` adds no value, and for that I would simplify it to:

```suggestion
        for (j = 0; j < messages.length; j++) {
```

Reply:
Whether to invoke `length` in a for loop is a longstanding debate. (ex:  https://stackoverflow.com/questions/17989270/for-loop-performance-storing-array-length-in-a-variable) I think it would be fine in this case, but opted for the existing code style for this project.

Reply:
Interesting point, @waltjones. However, I wouldn't trust a random StackOverflow thread. Keep in mind that all modern web browsers have implemented JIT and one of the JS JIT principles is loop expansion, so IMHO there is nothing to worry about. Of course, in some random tests, the results may vary. There are also some safe guards around the recompiled code, so in some scenarios it's possible to prove that _"jittable"_ version is _"slower"_. At least that was truth about ~ 10 yrs ago, when we were implementing PyPy JIT based on JavaScript JIT principles.

Reply:
Bart, that's an antagonistic thing to say. I'm not trusting a random thread. I stated clearly that I followed the style of the current project, not a random internet thread.

For example,
https://github.com/rollbar/rollbar.js/blob/v2.21.1/src/predicates.js#L62-L63
and
https://github.com/rollbar/rollbar.js/blob/v2.21.1/src/predicates.js#L102-L103

in the same file as this change.

Reply:
Oh, sorry Walt. I totally agree that it's important to keep consistent style across the project. So, in fact, I would rather stick with your previous version.

Since the PR has already been approved and merged, I just wanted to comment on the general question that was raised in the StackOverflow thread (because I found this debate interesting). I didn't mean to say you're relying on random SO threads. I have your point that this has been debated for a long time and I just wanted to put my two cents into the debate itself. I referred to general quality of approved answers on StackOverflow that sometimes are not the best answers though.

I'm sorry for the misunderstanding!

Reply:
:+1: Thank you, and no worries.